packages/components/src/tab-panel/index.tsx

```diff
@@ -106,7 +106,7 @@ export function TabPanel( {
 	const selectedId = `${ instanceId }-${ selectedTab?.name ?? 'none' }`;
 
 	useEffect( () => {
-		if ( ! selectedTab?.name && tabs.length > 0 ) {
```

Comment:
⚠️ Do not commit this!

The `TabPanel` is an uncontrolled component. `initialTabName` prop is only for setting the initial tab and from then on the component manages its own state internally. 

However, we now need a way to set the internal state from the consumer. There are various ways of achieving such a thing but I will reach out to the folks working on `@wordpress/components` for advice.

Likely this will be pulled out into a separate PR.

Reply:
One option is I guess to make `selectedTab` optional. If provided the component is expected to be "controlled" and you have to handle the `selectedTab` state as the consumer.